v2_x/git-remote-gosh/src/git_helper/push/mod.rs

```diff
@@ -279,6 +282,69 @@ where
         ))
     }
 
+    #[instrument(level = "trace")]
+    async fn find_ancestor_commit(
+        &self,
+        from: git_repository::Id<'_>,
+    ) -> anyhow::Result<Option<String>> {
+        let walk = from
+            .ancestors()
+            .all()?
+            .map(|e| e.expect("all entities should be present"))
+            .into_iter();
+
+        let accounts: Vec<_> = walk.map(|a| a.to_string()).collect();
+
+        let query = r#"query($accounts: [String]!) {
+            accounts(filter: {
```

Comment:
Accounts can be not sorted here

Reply:
It definitely won't be done, but the subsequent search is performed over a sorted list of commits